pasqal_cloud/__init__.py

```diff
@@ -234,44 +236,115 @@ def cancel_batch(self, id: str) -> Batch:
 
         Raises:
             RebatchError if rebatch call failed.
         """
+        if filters is None:
+            filters = RebatchFilters()
+        elif not isinstance(filters, RebatchFilters):
+            raise ValueError(
+                f"Filters needs to be a RebatchFilters instance, not a {type(filters)}"
+            )
+
         try:
-            new_batch_data = self._client.rebatch(
-                id,
-                job_ids=job_ids,
-                status=status,
-                min_runs=min_runs,
-                max_runs=max_runs,
-                start_date=start_date,
-                end_date=end_date,
+            new_batch_data = self._client._rebatch(
+                batch_id=id,
+                filters=filters,
             )
         except HTTPError as e:
             raise RebatchError(e) from e
         return Batch(**new_batch_data, _client=self._client)
 
+    def get_jobs(
+        self,
+        filters: Optional[JobFilters] = None,
+        from_index: int = 0,
+        page_size: int = 100,
+    ) -> dict[str, Union[int, List[Job]]]:
+        """
+        Retrieve a list of jobs matching filters using a pagination system.
+
+        By default, it returns the first 100 jobs out of the total of jobs
+        matching the query.
+
+        Args:
+            filters: filters to be applied to the jobs
+            from_index: The starting index of the jobs to retrieve.
+            page_size: The number of jobs to retrieve per page.
+
+        Examples:
+            >>> get_jobs(from_index=0, page_size=100,
+            JobFilters(status=JobStatus.ERROR))
+            This will return the first 100 jobs with an ERROR status.
+            >>> get_jobs(from_index=100, page_size=100,
+            JobFilters(status=JobStatus.ERROR))
+            This will return the next 100 jobs with an ERROR status.
+
+        Returns:
+            A dictionary with two keys:
+            - total: The total number of Jobs matching the filters.
+            - results: A list of Jobs matching the filters and
+                       pagination parameters provided.
+
+        Raises:
+            JobFetchingError: If fetching jobs call failed.
+        """
+        if not isinstance(from_index, int):
+            raise ValueError(
+                "Wrong type for 'from_index', it needs to be an integer, "
+                f"not a {type(filters)}"
+            )
+
+        if from_index < 0:
+            raise ValueError(
+                "Wrong value for 'from_index', it has to be a positive integer"
+            )
+
+        if not isinstance(page_size, int):
+            raise ValueError(
+                "Wrong type for 'page_size', it needs to be an integer, "
+                f"not a {type(filters)}"
+            )
+
+        if not (1 <= page_size <= 100):
+            raise ValueError(
+                "Wrong value for 'page_size', it has to be a positive integer "
+                "between 1 and 100"
+            )
```

Comment:
We will implement pagination for `get_batch` as well, which will have to perform the same checks. Let's introduce a class to regroup this validation logic